src/Bot/Dubtrack.hs

```diff
@@ -63,17 +69,50 @@ songLink (songType -> SongTypeSoundcloud) =
   "Soundcloud links are not supported yet"
 songLink _ = error "This should never happen Kappa"
 
+newtype RoomName = RoomName
+  { unName :: T.Text
+  }
+
+instance IsEntity RoomName where
+  nameOfEntity _ = "DubtrackRoom"
+  toProperties reply = Map.fromList [("name", PropertyText $ unName reply)]
+  fromProperties = fmap RoomName . extractProperty "name"
+
+getRoom :: Effect (Maybe (Entity RoomName))
+getRoom = listToMaybe <$> selectEntities P.Proxy (Take 1 All)
+
+setDubtrackRoom :: Reaction Message T.Text
+setDubtrackRoom =
+  liftR
+    (\msg -> do
+       mayReply <- getRoom
+       case mayReply of
+         Just reply ->
+           void $ updateEntityById $ (\a -> a {unName = msg}) <$> reply
+         Nothing -> void $ createEntity P.Proxy $ RoomName msg) $
+  cmapR (const "Updated room for dubtrack") $ Reaction replyMessage
+
 -- TODO(#221): Dubtrack room is hardcode
 currentSongCommand :: Reaction Message ()
 currentSongCommand =
   Reaction $ \Message {messageSender = sender} -> do
-    request <- parseRequest "https://api.dubtrack.fm/room/tsoding"
-    response <- eitherDecode . getResponseBody <$> httpRequest request
-    case response of
-      Left message -> errorEff $ T.pack message
-      Right dubtrackResponse ->
-        maybe
-          (replyToSender sender "Nothing is playing right now")
-          (\song ->
-             replyToSender sender [qms|❝{songName song}❞: {songLink song}|])
-          (roomCurrentSong $ drData dubtrackResponse)
+    mayRoom <- getRoom
+    case mayRoom of
+      Nothing ->
+        replyToSender
+          sender
+          "Dubtrack room not set, a mod can run '!config dubtrack $ROOM_NAME' to set it"
+      Just mahroom -> do
+        request <-
+          parseRequest $
+          "https://api.dubtrack.fm/room/" <>
+          T.unpack (unName $ entityPayload mahroom)
+        response <- eitherDecode . getResponseBody <$> httpRequest request
+        case response of
+          Left message -> errorEff $ T.pack message
+          Right dubtrackResponse ->
+            maybe
+              (replyToSender sender "Nothing is playing right now")
+              (\song ->
+                 replyToSender sender [qms|❝{songName song}❞: {songLink song}|])
+              (roomCurrentSong $ drData dubtrackResponse)
```

Comment:
The Reaction type was specifically introduced to simply these long case-chains. Please add a TODO to rewrite this using the Reaction type and I will look into that later. See https://github.com/tsoding/snitch/#unreported-todo for more info on how to format a TODO.

Reply:
I tried doing it myself for a bit but got stuck, I don't really know how to rewrite this into that in a constructive manner. Eg I don't know where to start.

Reply:
@jappeace we can try to do that on the next Thursday stream, how about that?

Reply:
Excelent!